enhancements/network/on-prem-bgp-networking.md

```diff
@@ -0,0 +1,455 @@
+The customer would update the BGP Configuration via the OpenShift API, and baremetal-runtimecfg which would monitor the Infrastructure object would update the configuration of the control plane nodes and reload FRR.
+
+### Risks and Mitigations
+
+- This design heavily relies on the infrastructure network that is not controlled by OpenShift. Documentation and validations will be provided
+  to ensure that prerequisites are met before a deployment can be done using BGP
+- Performance can be degraded if infrastructure network has routing problems, but routing issues would also involve a general outage, not just control plane VIPs. To a certain extent, this is out of our control,
+  but we'll provide documentation and tools to help the customer to configure the infrastructure network correctly and also run FRR healthchecks that will alert the customer if something is wrong with the BGP sessions.
+- The opinionated configuration for FRR might be too simple for now, but we'll adjust it as we get feedback from customers and field deployments
+
+### Drawbacks
+
+The proposed solution heavily relies on the network infrastructure at the data center level and if something goes wrong
+with BGP, the OpenShift cluster can become very unstable.
+Also, the solution remains complex to implement and corner cases will have to be documented and covered by testing when possible (e.g. day 2 operations).
+
+## Design Details
+
+### Test Plan
+
+Testing will be implemented for OpenStack at first and re-usable by other platforms.
+We'll provide a set of scripts that can install ToR routers and this will be used to replicate
+the network infrastructure that customers could run.
+The jobs will do the following:
+- Create a proxy node that runs FRR and will be the upstream router for the OpenShift cluster
+- Deploy the OpenShift cluster and provides an `install-config.yaml` that has the needed options for BGP
+- Run the desired e2e test suite
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+These following requirements will apply first for OpenStack, and once this has proven to be stable, we'll turn this on for other platforms:
+* The code has been merged into various components (API, installer, MCO, baremetal-runtimecfg)
+* We have an image for FRR
+* Feature is documented
+* Feature is tested in CI
+
+Once these requirements are met, then we'll deliver this feature as Tech Preview. The current desired target is OCP 4.13.
+
+Note: OpenStack 16.1 or 16.2 or 17.1 are the recommended versions to run at the moment.
+
+#### Tech Preview -> GA
+
+To go GA, this feature will have to be proven to be stable enough and tested by our QE.
+The current desired target is OCP 4.14.
+
+#### Removing a deprecated feature
+
+### Upgrade / Downgrade Strategy
+
+If FRR configuration file has to change, we'll have a monitor container that will apply the new configuration and reload FRR daemons.
```

Comment:
We don't plan to offer a migration path from L2-based load-balancing to a BGP-based one.